src/index.ts

```diff
@@ -785,9 +880,50 @@ export default class MapboxPathControl implements IControl {
       }
       this.referencePoints.splice(this.selectedReferencePointIndex!, 1);
       this.linesBetweenReferencePoints.splice(nextLine.properties!.index, 1);
-      this.syncIndex();
+      if (this.referencePoints.length < 3 && this.isLoopTrail) {
```

Comment:
Can you add a comment here to explain what you're doing ?

Reply:
Done 👍